src/AspNetCoreModuleV2/InProcessRequestHandler/StartupExceptionApplication.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+#include "StartupExceptionApplication.h"
+
+VOID StartupExceptionApplication::ShutDown()
+{
+    if (!m_pHttpServer->IsCommandLineLaunch())
+    {
+        // IIS scenario.
+        // notify IIS first so that new request will be routed to new worker process
+        m_pHttpServer->RecycleProcess(L"AspNetCore InProcess Recycle Process on Demand");
+    }
+    else
+    {
+        // IISExpress scenario
+        // Shutdown the managed application and call exit to terminate current process
+        exit(0);
```

Comment:
did you copy it from somewhere? We should avoid doing exit(0) and allow IIS to gracefully stop.

Reply:
Yeah we shouldn't need it at all.